executor/showtest/show_test.go

```diff
@@ -458,6 +458,26 @@ func TestShowCreateTable(t *testing.T) {
 			"  `b` varchar(20) DEFAULT '\\\\',\n"+
 			"  PRIMARY KEY (`a`) /*T![clustered_index] CLUSTERED */\n"+
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
+
+	// Test tidb_enable_simplified_show_create_table
```

Comment:
Do we need to add more tests for `AUTO_ID_CACHE`, `PLACEMENT POLICY` , `AUTO_RANDOM_BASE` , `SHARD_ROW_ID_BITS` ...

Reply:
I will add one test for each, just to verify that it still works.
In general I think we should have more tests of show create table syntax, at least to make sure the output is valid input, like the boarder line case of https://github.com/pingcap/tidb/issues/34262

Reply:
Added tests.